devel/fortio/http.go

```diff
@@ -15,34 +15,50 @@
 package fortio
 
 import (
+	"flag"
 	"fmt"
+	"io"
 	"io/ioutil"
-	"log"
 	"net"
 	"net/http"
 	"net/http/httputil"
+	"net/url"
 	"strings"
 	"time"
+	"unicode/utf8"
 )
 
-// ExtraHeaders to be added to each request.
-var extraHeaders http.Header
+// Fetcher is the Url content fetcher that the different client implements.
+type Fetcher interface {
+	// Fetch returns http code, data, offset of body (for client which returns
+	// headers)
+	Fetch() (int, []byte, int)
+}
 
-// Host is treated specially, remember that one separately.
-var hostOverride string
+var (
+	// ExtraHeaders to be added to each request.
+	extraHeaders http.Header
+	// Host is treated specially, remember that one separately.
+	hostOverride string
+	// Buffer size for optimized client
+	bufferSizeKbFlag                = flag.Int("httpbufferkb", 32, "Size of the buffer (max data size) for the optimized http client in kbytes")
```

Comment:
it's sort of a constant, but configurable, but I can move it in the constructor but it'll more more hassles to transmit it